src/java/picard/sam/FastqToSam.java

```diff
@@ -169,36 +175,105 @@ public static void main(final String[] argv) {
         System.exit(new FastqToSam().instanceMain(argv));
     }
 
+    /**
+     * Get a list of FASTQs that are sequentially numbered based on the first (base) fastq.
+     * The files should be named:
+     *   <prefix>_001.<extension>, <prefix>_002.<extension>, ..., <prefix>_XYZ.<extension>
+     * The base files should be:
+     *   <prefix>_001.<extension>
+     * An example would be:
+     *   RUNNAME_S8_L005_R1_001.fastq
+     *   RUNNAME_S8_L005_R1_002.fastq
+     *   RUNNAME_S8_L005_R1_003.fastq
+     *   RUNNAME_S8_L005_R1_004.fastq
+     * where `baseFastq` is the first in that list.
+     */
+    protected static List<File> getSequentialFileList(final File baseFastq) {
+        final List<File> files = new ArrayList<File>();
+        files.add(baseFastq);
+
+        // Find the correct extension used in the base FASTQ
+        FastqExtensions fastqExtensions = null;
+        String suffix = null; // store the suffix including the extension
+        for (final FastqExtensions ext : FastqExtensions.values()) {
+            suffix = "_001" + ext.getExtension();
+            if (baseFastq.getAbsolutePath().endsWith(suffix)) {
+                fastqExtensions = ext;
+                break;
+            }
+        }
+        if (null == fastqExtensions) {
+            throw new PicardException(String.format("Could not parse the FASTQ extension (expected '_001' + '%s'): %s", FastqExtensions.values().toString(), baseFastq));
+        }
+        
+        // Find all the files
+        for (int idx = 2; true; idx++) {
+            String fastq = baseFastq.getAbsolutePath();
```

Comment:
Seems like this would be easier using htsjdk IOUtil

see getFilesMatchingRegexp